src/Server/src/Player.java

```diff
@@ -72,60 +77,157 @@ public void run() {
         this.playerName = playerName;
     }
 
+    /**
+     *
+     */
     private void disconnect() {
         up.disconnectPlayer(this);
         Thread.currentThread().interrupt();
     }
 
+    // DB management
+
+    /**
+     * Checks for user in the database, compares passwords
+     * Sends appropriate message to the client afterwards
+     *
+     * @param message String message which method uses
+     */
+    private void handleLogin(Message message) {
+        try {
+            String splitted[] = message.getStringMultipleContent();
+            String uName = splitted[0];
+            String uPasswd = encryptionProvider.encrypt(splitted[1]);
+
+            String query = "SELECT * FROM accounts WHERE nickname = '" + uName + "'";
+            ResultSet resultSet = databaseCommunicator.executeQuery(query);
+            if (!resultSet.isBeforeFirst()) {
+                messageProvider.sendMessage(new Message("alert", "Wrong username!"));
+                return;
+            }
+            resultSet.next();
+            if (resultSet.getString("password").equals(uPasswd)) {
+                messageProvider.sendMessage(new Message("login", true));
+                setPlayerName(uName);
+                setPlayerId(resultSet.getInt("user_id"));
+            } else {
+                messageProvider.sendMessage(new Message("alert", "Wrong password!"));
+            }
+        } catch (Exception e) {
+            messageProvider.sendMessage(new Message("alert", String.valueOf(e)));
+        }
+    }
+
+    /**
+     * Handles user registration process
+     * Sends appropriate message to the client afterwards
+     *
+     * @param message String message which method uses
+     */
+    private void handleRegister(Message message) {
+        try {
+            String splitted[] = message.getStringMultipleContent();
+            String uName = splitted[0];
+            String uPasswd = encryptionProvider.encrypt(splitted[1]);
+            String backupPin = splitted[2];
+            String email = splitted[3];
+
+            //Check if email already registered
+            String query = "SELECT * FROM accounts WHERE email = '" + email + "'";
+            ResultSet resultSet = databaseCommunicator.executeQuery(query);
+            if (resultSet.isBeforeFirst()) {
+                messageProvider.sendMessage(new Message("alert", "e-mail already registered!"));
+                return;
+            }
+
+            //Check for username availibility
+            query = "SELECT * FROM accounts WHERE nickname = '" + uName + "'";
+            resultSet = databaseCommunicator.executeQuery(query);
+            if (resultSet.isBeforeFirst()) {
+                messageProvider.sendMessage(new Message("alert", "Username taken!"));
+                return;
+            }
+
+            //Register user
+            //Get new user id
+            query = "SELECT user_id FROM accounts WHERE user_id = (SELECT max(user_id) FROM accounts)";
```

Comment:
Dwie linijki niżej to jest. Pobiera z bazy maksymalny UID, po czym dodaje do niego 1